src/js/utils/Select.jsx

```diff
@@ -115,9 +133,26 @@ class Select extends Component {
     }
 
     if (showLabelTooltip) {
-      const valueMapped = multi && value ?
-        this.props.value.map(v => v?.[labelKey] ?? v?.label) : [];
-      const valueLabel = multi ? valueMapped.join(', ') : (value.label || value.name);
+      let valueLabel = '';
+      if (value) {
```

Comment:
I understand your remark and I also prefer to use returns in `if(...)` statement, to short circuit the logic, but this is a little different case since in the end we need to return a html tag.

All I am doing in this code is assigning a  `valueLabel` that later will be used in the return below

https://github.com/openboxes/openboxes/blob/82af9198c5ca243772f846f9de3ff5723493def6/src/js/utils/Select.jsx#L156-L160

Reply:
we could always separate "label building" from this function and then just get in the `getTooltipHtml()` already the proper label like:
```javascript
const valueLabel = getValueLabel() ?? '';
```
this way we could inject the "return logic" proposed by Alan, but this comment is just as "a bonus".